### PR TITLE
Improved readability of ensure_bootstrapped

### PR DESCRIPTION
**Pull request by autrilla**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10620)

### CODE REVIEW

**Comment by Ms2ger** [comment]
Where did the `system-rust` support go?

**Comment by autrilla** [comment]
@Ms2ger I must have removed that by accident. It's back now.

File: python/servo/command_base.py [hunk 1]
```diff
@@ -405,18 +405,21 @@ def ensure_bootstrapped(self, targets=[]):
...
+        all_targets_exist = all([path.exists(p) for p in target_paths])
+
+        if (not self.config['tools']['system-rust'] and
+                (not rustc_binary_exists or not all_targets_exist)):
+            print("looking for rustc at %s" % (rustc_path))
             Registrar.dispatch("bootstrap-rust", context=self.context, target=targets)
-        if not self.config["tools"]["system-cargo"] and \
-           not path.exists(path.join(
-                self.config["tools"]["cargo-root"], "cargo", "bin", "cargo" + BIN_SUFFIX)):
+        if not self.config["tools"]["system-cargo"] and not rustc_binary_exists:
```
Comment (Wafflespeanut) [inline python/servo/command_base.py]: Why aren't we checking for cargo binary here?

Comment (autrilla) [inline python/servo/command_base.py]: Again, another mistake, sorry!

**Comment by Wafflespeanut** [comment]
Thanks! :)

@bors-servo r+

**Comment by KiChjang** [comment]
@bors-servo retry
* #10656, #10663

**Pull request merged**